modules/olmv1-creating-a-cluster-role.adoc

```diff
@@ -0,0 +1,344 @@
+// Module included in the following assemblies:
+//
+// * extensions/ce/managing-ce.adoc
+
+:_mod-docs-content-type: PROCEDURE
+
+[id="olmv1-creating-a-cluster-role_{context}"]
+= Creating a cluster role for an extension
```

Comment:
@michaelryanpeter I do not understand why only clusterole(binding) is created.

per https://operator-framework.github.io/operator-controller/howto/derive-service-account/, some of them could be created as role(binding).

by the way, https://github.com/operator-framework/operator-controller/blob/main/config/samples/olm_v1_clusterextension.yamlhas argocd-installer-role, argocd-installer-clusterrole, and argocd-installer-rbac-clusterrole. I am ok to combine argocd-installer-rbac-clusterrole and argocd-installer-clusterrole into one clusterrole as argocd-installer-clusterrole because they are clusterrole. I guess it is what you do in this PR.

Reply:
This was very difficult for me to do, and I had I to figure it out on my own. I did use the upstream docs as a guide though. 

When I looked at the CSV for the pipelines Operator, I could not tell which resources should be namespace scoped. I did notice the one instance of the `role` resource here: https://github.com/openshift/openshift-docs/pull/87627/files#diff-d6ea754a5b22ac27d3638ea281b18cbb03eedb0fd76abb2efe926684b6b382e8R161

I had to fail a lot of times to get permissions that would install. Please let me know if there is a better way to organize or break up the permissions: https://github.com/openshift/openshift-docs/pull/87627/files#diff-d6ea754a5b22ac27d3638ea281b18cbb03eedb0fd76abb2efe926684b6b382e8

Reply:
@michaelryanpeter frankly I have no better way to identify which is role and which is clusterrole. I think it is also difficult for customer. maybe query with SME (@joelanford ?) on it.  thanks

Reply:
@kuiwang02 Good spotting that some of the permissions can be scoped to just the install Namespace.
I've talked with @michaelryanpeter today and I advised to just add a notice for now that the example permissions could be restricted further and improve the fidelity of the examples in the next iteration.

This feature is _very_ complex to use as-is and I am not sure that further complicating the examples is worth it at this time. @perdasilva has already started tinkering with a tool to auto-generate these permissions and ease the burden on the administrator significantly.

Additionally we are going to have to revisit this doc anyway with the upcoming Single/OwnNamespace install-mode support in mind, because that feature also impacts how much RBAC can be restricted.

Reply:
Added an admonition to clarify that the examples rely on cluster roles for example purposes in this commit: https://github.com/openshift/openshift-docs/pull/87627/commits/7cee26fbeaf56af6ea948a3862d3e31af60ffeea

Reply:
@kuiwang02 Would you please take another look?

Reply:
@thetechnick @michaelryanpeter Thanks. 

/lgtm